src/query/cypher_query_interpreter.cpp

```diff
@@ -128,27 +129,21 @@ std::unique_ptr<LogicalPlan> MakeLogicalPlan(AstStorage ast_storage, CypherQuery
 }
 
 std::shared_ptr<CachedPlan> CypherQueryToPlan(uint64_t hash, AstStorage ast_storage, CypherQuery *query,
-                                              const Parameters &parameters, utils::SkipList<PlanCacheEntry> *plan_cache,
+                                              const Parameters &parameters, PlanCacheLRU *plan_cache,
                                               DbAccessor *db_accessor,
                                               const std::vector<Identifier *> &predefined_identifiers) {
-  std::optional<utils::SkipList<PlanCacheEntry>::Accessor> plan_cache_access;
   if (plan_cache) {
-    plan_cache_access.emplace(plan_cache->access());
-    auto it = plan_cache_access->find(hash);
-    if (it != plan_cache_access->end()) {
-      if (it->second->IsExpired()) {
-        plan_cache_access->remove(hash);
-      } else {
-        return it->second;
-      }
+    if (plan_cache->WithLock([&](auto &cache) { return cache.exists(hash); })) {
+      return plan_cache->WithLock([&](auto &cache) { return cache.get(hash); });
```

Comment:
Since exists is combined with get, we can maybe have function, try_get, which returns if exists, otherwise returns nullptr, so instead of taking lock twice, we take once